server/pbench/bin/pbench-server-prep-shim-002

```diff
@@ -0,0 +1,175 @@
+#! /bin/bash
+# set -x
+
+# This shim is used to prepare the tarballs that a version 002 client
+# submits for further processing. It copies the tarballs and their MD5
+# sums to the archive (after checking) and sets the state links, so
+# that the dispatch script will pick them up and get the ball
+# rolling. IOW, it does impedance matching between version 002 clients
+# and the server scripts.
+
+###########################################################################
+# load common things
+opts=$SHELLOPTS
+case $opts in
+    *xtrace*)
+        dir=$(dirname $(which $0))
+        PROG=$(basename $(which $0))
+        ;;
+    *)
+        dir=$(dirname $0)
+        PROG=$(basename $0)
+        ;;
+esac
+
+. $dir/pbench-base.sh
+###########################################################################
+
+archive=$(getconf.py archive-dir pbench-server)
+qdir=$(getconf.py quarantine-dir pbench-server)
+quarantine=${qdir}/md5-002
+duplicates=${qdir}/duplicates-002
+
+# The following directory holds tarballs that are quarantined because
+# of operational errors on the server. They should be retried after
+# the problem is fixed: basically, move them back into the reception
+# area for 002 agents and wait.
+errors=${qdir}/errors-002
+
+errlog=$LOGSDIR/$PROG/$PROG.error
+if [[ "${archive}" == "" ]] ;then
+    echo "Failed: \"getconf.py archive-dir pbench-server" >> $errlog
+    exit 1
+fi
+
+if [[ "$quarantine" == "" ]] ;then
+    echo "Failed: \"getconf.py quarantine-dir pbench-server" $errlog
+    exit 1
+fi
+
+if [[ "$duplicates" == "" ]] ;then
+    echo "Failed: \"getconf.py duplicates-dir pbench-server" $errlog
+    exit 1
+fi
+
+log_init $(basename $0) $LOGSDIR/$(basename $0)
+
+tmp=$TMP/$PROG.$$
+
+trap 'rm -rf $tmp' EXIT INT QUIT
+
+mkdir -p $tmp
+list=$tmp/list.check
+status=$tmp/status
+> $status
+results_dir=$(getconf.py receive-dir-002 pbench-server)
+# Check for results that are ready for processing: version 002 agents
+# upload the MD5 file as xxx.md5.check and they rename it to xxx.md5
+# after they are done with MD5 checking so that's what we look for.
+find ${results_dir} -maxdepth 2 -name *.tar.xz.md5 > $list
+
+typeset -i ntbs=0
+typeset -i nerrs=0
+typeset -i nquarantined=0
+typeset -i ndups=0
+
+echo $TS
+while read ck ;do
+    # full pathname of tarball
+    tb=${ck%.md5}
+
+    # directory
+    tbdir=$(dirname ${tb})
+
+    # resultname: get the basename foo.tar.xz and then strip the .tar.xz
+    resultname=$(basename ${tb})
+    resultname=${resultname%.tar.xz}
+
+    # the controller hostname is the last component of the directory part of the full path
+    controller=$(basename ${tbdir})
+
+    dest=${archive}/${controller}
+
+    # sanity checks
+    if [ ! -f ${tb} ] ;then
+        echo "$TS: ${tb} file does not exist" >&4
+        if [ -f ${tb}.md5 ] ;then
+            quarantine ${quarantine}/${controller} ${tb}.md5
+        fi
+        nquarantined=$nquarantined+1
```

Comment:
Should we only count quarantines when we call the `quarantine` function?

Reply:
Moved the counting inside the if (but as mentioned before, not every call to quarantine increments the quarantine count).